### ISSUE TITLE
Remove commit bit from inactive core devs as part of SC elections

### ISSUE THREAD

**Issue by hugovk** [python/steering-council#254]
During the [Language Summit 2024](https://us.pycon.org/2024/events/language-summit/) one thing we discussed regarding ["Strengthening Python's Security Model"](https://pyfound.blogspot.com/2024/06/python-language-summit-2024-python-security-model-after-xz.html) was removing the commit bit for inactive core devs.

Indeed, [PEP 13](https://peps.python.org/pep-0013/#membership) already has provisions for this:

>  Those who haven’t made any non-trivial contribution in two years may be asked to move themselves to [the “inactive”] category, and moved there if they don’t respond. [...] While someone is in inactive status, though, they lose their active privileges like voting or nominating for the steering council, and commit access.

We further discussed this in https://github.com/python/core-workflow/issues/539, and I'd like to suggest we start doing this as part of the annual election process.

I propose a process along the lines of:

1. As part of compiling the voter roll for the annual SC election, find inactive committers.
2. Contact them and invite them to become Emeritus committers and remove the commit bit for security. Let them know they can have it re-enabled in the future if they wish to become active. 
3. For those that don't reply after two weeks, or those that reply saying they wish to become Emeritus, add them to the Emeritus group and remove their commit bit.
4. Repeat annually after each SC election voter roll has been compiled.

When contacting inactive people, it's important to let them know they've not done anything wrong, it can be re-instated if they become active again, and thank them for their contribution (see https://github.com/python/core-workflow/issues/539#issuecomment-2152637975).

Please could the Steering Council consider this, starting with the upcoming 2025 term election?

**Comment by sobolevn** [comment]
> any non-trivial

I think that this needs a bit of more context.
Like: is making a commit a trivial contribution? What about a review / design discussion?

**Comment by hugovk** [comment]
> Like: is making a commit a trivial contribution?

You can see the scripts used for the elections at 🔒 https://github.com/python/voters. It looks for _any_ commit to CPython, which I think is okay for practicality reasons, to avoid having someone sit down and review commits for undefined "triviality".

> What about a review / design discussion?

I believe this is captured by the next step, asking pending inactive people if they wish to remain active. It also covers those who have committed to non-CPython https://github.com/python repos, and indeed any of the [valuable non-code core team contributions](https://peps.python.org/pep-0013/#membership).

**Comment by emilyemorehouse** [comment]
Thanks for raising this! Overall, the SC feels that this is a good addition to the process but will likely defer this decision to the next SC.

In the meantime, we'd like to suggest opening a discussion on DPO to gather community feedback on whether we should split out a Core Developer's status to support having voting rights but not commit rights. We can see a situation where someone would want to opt-in to being "active" in order to participate in votes, but don't actually need the commit bit.

**Comment by hugovk** [comment]
Thanks, I've opened a discussion at https://discuss.python.org/t/split-out-voting-rights-from-commit-rights-for-inactive-core-members/68208.

**Comment by gpshead** [comment]
SC discussion status: internal discussion ongoing - no updates to share yet.

**Comment by warsaw** [comment]
@hugovk We just had our first meeting of the 2026 SC.  You have the 👍 to make this happen.

**Comment by warsaw** [comment]
@hugovk What is the status of this?  Do you need anything from the SC?  Asking so I can update the SC at tomorrow's (Thursday's) meeting.

**Comment by hugovk** [comment]
Yes, from the 22nd January office hours, the SC was going to comment on this issue about the plan, then I was going to do it :)

From my notes, the plan was:

* We do want the the active status to be granular in python/voters, for example, so someone can express: "I want to vote, but not commit"
* SC asked me to send the initial email to the first round of removals, then wait a month, and send a reminder

Please could the SC confirm/correct/expand upon this as necessary? Thanks!

**Comment by Yhg1s** [comment]
Yes, that's the plan we discussed. I think we suggested sending the second email a week or so before actually removing people.

**Comment by gpshead** [comment]
poke. can we do this?

**Comment by hugovk** [comment]
Yes, not forgotten about it, I'm planning to get back to this next month.

**Comment by corona10** [comment]
Hi @hugovk 

SC has unanimously agreed to the plan you outlined above. 
Please go ahead with the first group in https://github.com/python/voters/pull/122.
For the other two groups, bring them back to the SC if you'd like us to weigh in.

**Comment by hugovk** [comment]
Thanks! I'll plan this for September.

**Comment by hugovk** [comment]
First group emailed, see https://github.com/python/voters/pull/125.

Plan:

* [x] 2026-09-11: open PR of current active committers: [python/voters#125](https://github.com/python/voters/pull/125)
* [x] 2026-09-11: send emails to inactive
* [ ] 2026-10-11: wait a month (update `python-core.toml` as replies come in: [python/voters#126](https://github.com/python/voters/pull/126))
* [ ] 2026-10-11: send a reminder to any non-responders
* [ ] 2026-10-18: a week or so later, remove inactive (update `python-core.toml` and co-ordinate with org owner to update GH teams)
